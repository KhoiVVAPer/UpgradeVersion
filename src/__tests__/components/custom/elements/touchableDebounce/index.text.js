import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TouchableDebounce from '../../../../../components/custom/elements/touchableDebounce';

const setUp = (props) => {
  const component = shallow(<TouchableDebounce {...props} />);
  return component;
};


describe('components > custom > elements > touchableDebounce', () => {
  let component;
  beforeEach(() => {
    const props = {
    };
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders Text component', () => {
    const input = component.find('TouchableOpacity').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Check if text component contain onPress prop', () => {
    const input = component.find('TouchableOpacity').prop('onPress');
    expect(input).toBeDefined();
  });
});
