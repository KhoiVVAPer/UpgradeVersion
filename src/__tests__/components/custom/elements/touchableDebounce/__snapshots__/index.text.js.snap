// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > elements > touchableDebounce Check snapshot 1`] = `<react-native-mock />`;
